pkg/core/blockchain.go

```diff
@@ -1204,9 +1166,19 @@ func (bc *Blockchain) verifyTx(t *transaction.Transaction, block *block.Block) e
 	if t.ValidUntilBlock <= height || t.ValidUntilBlock > height+transaction.MaxValidUntilBlockIncrement {
 		return errors.Errorf("transaction has expired. ValidUntilBlock = %d, current height = %d", t.ValidUntilBlock, height)
 	}
+	balance := util.Fixed8FromInt64(bc.GetNEP5Balances(t.Sender).Trackers[bc.contracts.GAS.Hash].Balance)
+	need := t.SystemFee.Add(t.NetworkFee)
+	if balance.LessThan(need) {
+		return errors.Errorf("insufficient funds: balance is %v, need: %v", balance, need)
+	}
 	if io.GetVarSize(t) > transaction.MaxTransactionSize {
 		return errors.Errorf("invalid transaction size = %d. It shoud be less then MaxTransactionSize = %d", io.GetVarSize(t), transaction.MaxTransactionSize)
 	}
+	netFee := t.NetworkFee.FloatValue() - float64(io.GetVarSize(t))*bc.FeePerByte(t).FloatValue()
+	needNetworkFee := float64(io.GetVarSize(t)) * bc.FeePerByte(t).FloatValue()
```

Comment:
Calculating these in `float64` is a bad idea.

Reply:
Fixed with calculating in `int64` and `(Fixed8).IntegralValue()`